syntax/printer.go

```diff
@@ -269,16 +269,17 @@ func (p *Printer) space() {
 }
 
 func (p *Printer) spacePad(pos Pos) {
-	if p.cols.lineStart {
-		// Never add padding at the start of a line, since this may
+	if p.cols.lineStart && (!p.keepPadding || p.indentSpaces == 0) {
+		// Never add padding at the start of a line unless keepPadding
+		// is set and we are indenting with spaces, since this may
 		// result in broken indentation or mixing of spaces and tabs.
 		return
 	}
-	if p.wantSpace {
+	if p.wantSpace && !p.cols.lineStart {
```

Comment:
hmm, this feels backwards; if anything calls spacePad at the start of a line, then wantSpace should be false. If you don't know how to fix that, that's fine, you can leave a TODO here and I'll look into it at a later point.

Reply:
No you are right, this additional check was useless